### ISSUE TITLE
Autovalue bug when inserting into array

### ISSUE THREAD

**Issue by benlavalley** [aldeed/meteor-collection2-core#14]
The example for keeping updates history for a field appears as though it is longer working with simpl-schema 0.1.0 and Collection2-core 2.0.0. 

From [https://github.com/aldeed/meteor-collection2-core/](): 
```
// Whenever the "content" field is updated, automatically
  // update a history array.
  updatesHistory: {
    type: Array,
    optional: true,
    autoValue: function() {
      var content = this.field("content");
      if (content.isSet) {
        if (this.isInsert) {
          return [{
              date: new Date(),
              content: content.value
            }];
        } else {
          return {
            $push: {
              date: new Date,
              content: content.value
            }
          };
        }
      } else {
        this.unset();
      }
    }
  },
  'updatesHistory.$': {
    type: Object,
  },
  'updatesHistory.$.date': {
    type: Date,
    optional: true
  },
  'updatesHistory.$.content': {
    type: String,
    optional: true
  },
```

When using this in a collection it appears that autovalue unnecessarily creates an extra array doing the insert, yielding the error in my sample app:

`titleUpdatesHistory.0.0 is not allowed by the schema`

Example project here:

https://github.com/benlavalley/autoValueBug

Tried to put together as succinct an example as possible. I could dig into simpl-schema/collection2-core code more but I spent a number of hours working on this one already (moving from simpl-schema 0.0.3 to 0.1.0, adding babel-polyfill to my app for it, etc) and I think the right person will be able to swiftly figure this one out :)

Not sure if this belonged in simpl-schema or not - if it did let me know and I can create an issue there.

If I'm fouling something up or missed seeing a change in the new simpl-schema/collection2 versions I apologize - these packages are awesome and I'd hate to know I was polluting the issues section with something that was my fault.

**Comment by chathuraa** [comment]
Im having a similar issue.
` phoneNumbers: {
    type: Array,
    optional: true,
    autoValue() {
      const phone = this.field('primaryContactNo').value;
      return { $addToSet: phone };
    },
  },`

This  adds/update the array. however I get the following error.
`update failed: Error: phoneNumbers.0.0 is not allowed by the schema
    at getErrorObject (http://localhost:3000/packages/aldeed_collection2-core.js?hash=0564817159bbe9f8209b6a192a1fa2b2bbab924a:523:15)
    at _0x19bfx8.(anonymous function).doValidate (http://localhost:3000/packages/aldeed_collection2-core.js?hash=0564817159bbe9f8209b6a192a1fa2b2bbab924a:502:13)
    at _0x19bfx8.(anonymous function).Mongo.Collection.(anonymous function) [as update] (http://localhost:3000/packages/aldeed_collection2-core.js?hash=0564817159bbe9f8209b6a192a1fa2b2bbab924a:260:25)
    at DDPCommon.MethodInvocation.contacts.update (http://localhost:3000/app/app.js?hash=c0a0f05fa25b4c8b5ec03ecd3e6a0b6b93dd5e21:10720:21)
    at http://localhost:3000/packages/ddp-client.js?hash=c9ca22092a3137a7096e8ab722ba5ab8eedb9aec:4076:25
    at Meteor.EnvironmentVariable.withValue (http://localhost:3000/packages/meteor.js?hash=27829e936d09beae3149ecfbf3332c42ccb1596f:1077:17)
    at Connection.apply (http://localhost:3000/packages/ddp-client.js?hash=c9ca22092a3137a7096e8ab722ba5ab8eedb9aec:4067:54)
    at Connection.call (http://localhost:3000/packages/ddp-client.js?hash=c9ca22092a3137a7096e8ab722ba5ab8eedb9aec:3948:17)
    at ContactsEdit.handleOnSuccess (http://localhost:3000/app/app.js?hash=c0a0f05fa25b4c8b5ec03ecd3e6a0b6b93dd5e21:1022:14)
    at InlineEditInput.updateRecord (http://localhost:3000/app/app.js?hash=c0a0f05fa25b4c8b5ec03ecd3e6a0b6b93dd5e21:3763:18)`

**Comment by benlavalley** [comment]
I updated the sample project with the latest versions of collection2-core and simpl-schema and made a minor tweak to align as precisely as possible with the example in the documentation:

aldeed:collection2-core  - upgraded from 2.0.0 to 2.0.1
simpl-schema@0.1.1 - upgraded from simpl-schema@0.1.0
meteor and all packages - updated to 1.5.2.2

Unfortunately the problem still remains. 

The example here is highlighted as one of the premier features of collection2-core in the documentation, but it is still not working.

Schema here:
`titleUpdatesHistory: {
		type: Array,
		optional: true,
		autoValue() {
			const content = this.field('originalTitle');
				if (content.isSet) {
					if (this.isInsert) {
						return [{
							date: new Date(),
							content: content.value,
						}];
					} else {
						return {
							$push: { date: new Date(), content: content.value },
						};
					}
				} else {
					this.unset();
				}
		},
	},
	'titleUpdatesHistory.$': {
		type: Object,
	},
	'titleUpdatesHistory.$.date': {
		type: Date,
		optional: true,
	},
	'titleUpdatesHistory.$.content': {
		type: String,
		optional: true,
	},`

Yields this error upon updating the title:
![screen shot 2017-10-04 at 6 09 34 am](https://user-images.githubusercontent.com/7545485/31170597-a4204aee-a8ca-11e7-9e5f-027c455e2109.png)

I tried sifting through the source code to see if I could figure out the problem but didn't spot anything I could put together a reasonable fix for.
